### ISSUE TITLE
Report label in browse mode if it isn't rendered anywhere else (e.g. aria-label)

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4773]
**Reported by khsruru on 2015-01-05 07:56**
Hello, I’m Hyongsop Kim in Korea.
I have a suggestion about NVDA and aria-label.
Let’s suppose that one html code is blow.
`<input type=”text” aria-label=”type keyword” </input>`
In that case, if I press tab to approach to the edit box, NVDA reads aria-label attribute.
But in browse mode, if I press arrow key to approach, NVDA reads only edit, not read the aria-label text in Internet Explorer and Firefox.
In my opinion, aria-label text is used for the screen reader users in case not available to write visual text, NVDA should read the aria-label text even if use arrow keys.
Please check this issue.

Thank you.

**Comment by jcsteh** [comment]
P3 because this can mean the user misses information, but the implementation will take some time.

**Comment by derekriemer** [comment]
I think this is also seen when aria-labelledby is used but the idlist element is aria-hidden (The label is read by NVDA, but skipped in browse mode).

**Comment by fernando-jose-silva** [comment]
I find it particularly bad when navigating with nvda with the navigation mode with the arrows when there are two text boxes in succession, and the nvda in both reads only edit, and does not read the label, in this case to identify what is Requested in the form it is necessarily mandatory to press the nvda + tabe to know what to insert into each text box.To illustrate what I want to discuss, see the website:Http://www.3remetodista.org.br/When opening the page to get faster you can press the letter m twice to go to the second frame.Now navigate with arrows below until you reach the first text box, where nvda will only report editing, press another down arrow and nvda will arrive in the second text box where it will speak again editing.Note that since two text boxes one after the other can generate some confusion in not receiving the label from them, particularly when arriving in the second text box.The announcement of the labels when navigating with the navigation mode could be an option to be marked and unmarked by the user in the panel of the nvda.It is possible to obtain the information with nvda + tabe, however this command may not be known to the new users of nvda.This call took some directions that I did not understand either because I did not have a technical formation in programming, this my comment is more referring to the commentary that opened this call.

**Comment by michaelDCurran** [comment]
Note: In the VBufBackend for gecko, if explicitName and not nameIsContent and not isLabelVisible, then expose alwaysReportName as true.

**Comment by jcsteh** [comment]
Specific test case from @syntaxbliss in https://github.com/nvaccess/nvda/issues/5742#issue-133028610:

> I have the next markup:
> 
> `<button aria-label="LEFT BUTTON">left button</button><input type="text" aria-label="INPUT CONTROL" value="5"><button aria-label="RIGHT BUTTON">right button</button>`
> 
> When I'm standing on the left button and then press TAB, NVDA correctly reads the input aria-label, which is:
> 
> > INPUT CONTROL edit has autocomplete
> > selected 5
> 
> But if I'm standing on the left button and then press the arrow down key it reads:
> 
> > edit has autocomplete
> > 5

Note: This refers to pressing down arrow to get to the input field, which suggests screen layout was off for this test.
